src/web/services/user/join_discord.rs

```diff
@@ -93,9 +95,12 @@ pub async fn handle(auth: AuthenticationCookie) -> Result<HttpResponse, Telescop
         .ok_or(TelescopeError::ise("Could not get Verified role ID."))?;
 
     // Add user to Discord with verified role and nickname.
+    
+    let nickname_copy = &nickname;
     discord
-        .add_to_rcos_guild(Some(nickname), vec![verified_role])
+        .add_to_rcos_guild(Some(nickname_copy.to_string()), vec![verified_role])
         .await?;
+    
```

Comment:
Why are you copying the nickname here? It should be a string already? This seems unnecessary.

Reply:
Ownership issue. add_to_rcos_guild takes ownership of nickname so it couldn't be used later to change the nickname in the case that the user was already in the guild. I suppose an alternative would be to change add_to_rcos_guild to require a string reference rather than just a string, that can remove the need for making a copy of nickname.

Unless there's some third more elegant solution I'm not seeing?

Reply:
I think you should just be able to use `.clone()` on the nickname instead of creating a reference and then converting it separately